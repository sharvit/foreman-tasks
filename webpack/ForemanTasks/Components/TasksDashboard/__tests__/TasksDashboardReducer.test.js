import { testReducerSnapshotWithFixtures } from 'react-redux-test-utils';

import {
  FOREMAN_TASKS_DASHBOARD_INIT,
  FOREMAN_TASKS_DASHBOARD_UPDATE_TIME,
  FOREMAN_TASKS_DASHBOARD_UPDATE_QUERY,
  FOREMAN_TASKS_DASHBOARD_FETCH_TASKS_SUMMARY_REQUEST,
  FOREMAN_TASKS_DASHBOARD_FETCH_TASKS_SUMMARY_SUCCESS,
  FOREMAN_TASKS_DASHBOARD_FETCH_TASKS_SUMMARY_FAILURE,
} from '../TasksDashboardConstants';
import reducer from '../TasksDashboardReducer';

const fixtures = {
  'should return the initial state': {},
  'should handle FOREMAN_TASKS_DASHBOARD_INIT': {
    action: {
      type: FOREMAN_TASKS_DASHBOARD_INIT,
    },
  },
  'should handle FOREMAN_TASKS_DASHBOARD_INIT with data': {
    action: {
      type: FOREMAN_TASKS_DASHBOARD_INIT,
      payload: { time: 'some-time', query: 'some-query' },
    },
  },
  'should handle FOREMAN_TASKS_DASHBOARD_UPDATE_TIME': {
    action: {
      type: FOREMAN_TASKS_DASHBOARD_UPDATE_TIME,
      payload: 'some-time',
    },
  },
  'should handle FOREMAN_TASKS_DASHBOARD_UPDATE_QUERY': {
    action: {
      type: FOREMAN_TASKS_DASHBOARD_UPDATE_QUERY,
      payload: 'some-query',
    },
  },
  'should handle FOREMAN_TASKS_DASHBOARD_FETCH_TASKS_SUMMARY_REQUEST': {
    action: {
      type: FOREMAN_TASKS_DASHBOARD_FETCH_TASKS_SUMMARY_REQUEST,
    },
  },
  'should handle FOREMAN_TASKS_DASHBOARD_FETCH_TASKS_SUMMARY_SUCCESS': {
    action: {
      type: FOREMAN_TASKS_DASHBOARD_FETCH_TASKS_SUMMARY_SUCCESS,
      payload: 'some-payload',
    },
  },
  'should handle FOREMAN_TASKS_DASHBOARD_FETCH_TASKS_SUMMARY_FAILURE': {
    action: {
      type: FOREMAN_TASKS_DASHBOARD_FETCH_TASKS_SUMMARY_FAILURE,
      payload: new Error('some error'),
    },
  },
};

describe('BreadcrumbBar reducer', () =>
  testReducerSnapshotWithFixtures(reducer, fixtures));
