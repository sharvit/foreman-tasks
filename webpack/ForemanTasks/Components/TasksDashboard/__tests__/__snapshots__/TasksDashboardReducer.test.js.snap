// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BreadcrumbBar reducer should handle FOREMAN_TASKS_DASHBOARD_FETCH_TASKS_SUMMARY_FAILURE 1`] = `
Object {
  "error": [Error: some error],
  "isLoading": false,
  "isLoadingTasksSummary": false,
  "query": Object {},
  "tasksSummary": Object {
    "paused": Object {
      "recent": 0,
      "total": 0,
    },
    "running": Object {
      "recent": 0,
      "total": 0,
    },
    "scheduled": Object {
      "total": 0,
    },
    "stopped": Object {
      "by_result": Object {
        "error": Object {
          "recent": 0,
          "total": 0,
        },
        "success": Object {
          "recent": 0,
          "total": 0,
        },
        "warning": Object {
          "recent": 0,
          "total": 0,
        },
      },
    },
  },
  "time": "H24",
}
`;

exports[`BreadcrumbBar reducer should handle FOREMAN_TASKS_DASHBOARD_FETCH_TASKS_SUMMARY_REQUEST 1`] = `
Object {
  "error": null,
  "isLoading": true,
  "isLoadingTasksSummary": false,
  "query": Object {},
  "tasksSummary": Object {
    "paused": Object {
      "recent": 0,
      "total": 0,
    },
    "running": Object {
      "recent": 0,
      "total": 0,
    },
    "scheduled": Object {
      "total": 0,
    },
    "stopped": Object {
      "by_result": Object {
        "error": Object {
          "recent": 0,
          "total": 0,
        },
        "success": Object {
          "recent": 0,
          "total": 0,
        },
        "warning": Object {
          "recent": 0,
          "total": 0,
        },
      },
    },
  },
  "time": "H24",
}
`;

exports[`BreadcrumbBar reducer should handle FOREMAN_TASKS_DASHBOARD_FETCH_TASKS_SUMMARY_SUCCESS 1`] = `
Object {
  "error": null,
  "isLoading": false,
  "isLoadingTasksSummary": false,
  "query": Object {},
  "tasksSummary": "some-payload",
  "time": "H24",
}
`;

exports[`BreadcrumbBar reducer should handle FOREMAN_TASKS_DASHBOARD_INIT 1`] = `
Object {
  "error": null,
  "isLoadingTasksSummary": false,
  "query": Object {},
  "tasksSummary": Object {
    "paused": Object {
      "recent": 0,
      "total": 0,
    },
    "running": Object {
      "recent": 0,
      "total": 0,
    },
    "scheduled": Object {
      "total": 0,
    },
    "stopped": Object {
      "by_result": Object {
        "error": Object {
          "recent": 0,
          "total": 0,
        },
        "success": Object {
          "recent": 0,
          "total": 0,
        },
        "warning": Object {
          "recent": 0,
          "total": 0,
        },
      },
    },
  },
  "time": "H24",
}
`;

exports[`BreadcrumbBar reducer should handle FOREMAN_TASKS_DASHBOARD_INIT with data 1`] = `
Object {
  "error": null,
  "isLoadingTasksSummary": false,
  "query": "some-query",
  "tasksSummary": Object {
    "paused": Object {
      "recent": 0,
      "total": 0,
    },
    "running": Object {
      "recent": 0,
      "total": 0,
    },
    "scheduled": Object {
      "total": 0,
    },
    "stopped": Object {
      "by_result": Object {
        "error": Object {
          "recent": 0,
          "total": 0,
        },
        "success": Object {
          "recent": 0,
          "total": 0,
        },
        "warning": Object {
          "recent": 0,
          "total": 0,
        },
      },
    },
  },
  "time": "some-time",
}
`;

exports[`BreadcrumbBar reducer should handle FOREMAN_TASKS_DASHBOARD_UPDATE_QUERY 1`] = `
Object {
  "error": null,
  "isLoadingTasksSummary": false,
  "query": "some-query",
  "tasksSummary": Object {
    "paused": Object {
      "recent": 0,
      "total": 0,
    },
    "running": Object {
      "recent": 0,
      "total": 0,
    },
    "scheduled": Object {
      "total": 0,
    },
    "stopped": Object {
      "by_result": Object {
        "error": Object {
          "recent": 0,
          "total": 0,
        },
        "success": Object {
          "recent": 0,
          "total": 0,
        },
        "warning": Object {
          "recent": 0,
          "total": 0,
        },
      },
    },
  },
  "time": "H24",
}
`;

exports[`BreadcrumbBar reducer should handle FOREMAN_TASKS_DASHBOARD_UPDATE_TIME 1`] = `
Object {
  "error": null,
  "isLoadingTasksSummary": false,
  "query": Object {},
  "tasksSummary": Object {
    "paused": Object {
      "recent": 0,
      "total": 0,
    },
    "running": Object {
      "recent": 0,
      "total": 0,
    },
    "scheduled": Object {
      "total": 0,
    },
    "stopped": Object {
      "by_result": Object {
        "error": Object {
          "recent": 0,
          "total": 0,
        },
        "success": Object {
          "recent": 0,
          "total": 0,
        },
        "warning": Object {
          "recent": 0,
          "total": 0,
        },
      },
    },
  },
  "time": "some-time",
}
`;

exports[`BreadcrumbBar reducer should return the initial state 1`] = `
Object {
  "error": null,
  "isLoadingTasksSummary": false,
  "query": Object {},
  "tasksSummary": Object {
    "paused": Object {
      "recent": 0,
      "total": 0,
    },
    "running": Object {
      "recent": 0,
      "total": 0,
    },
    "scheduled": Object {
      "total": 0,
    },
    "stopped": Object {
      "by_result": Object {
        "error": Object {
          "recent": 0,
          "total": 0,
        },
        "success": Object {
          "recent": 0,
          "total": 0,
        },
        "warning": Object {
          "recent": 0,
          "total": 0,
        },
      },
    },
  },
  "time": "H24",
}
`;
